### ISSUE TITLE
High cpu usage with 0.12.7

### ISSUE THREAD

**Issue by Vorror** [vuejs/vetur#913]
- [X] I have searched through existing issues
- [X] I have read through [docs](https://vuejs.github.io/vetur)
- [X] I have read [FAQ](https://github.com/vuejs/vetur/blob/master/docs/FAQ.md)

## Info

- Platform: OSX
- Vetur version: 0.12.7
- VS Code version: 1.27.2

## Problem
I'm experiencing very high cpu usage when trying to use intellisense inside .vue files when using 0.12.7. Everything works correctly when I use the 0.12.1 vsix build directly from github.

Here's a gif with 0.12.1
As you can see everything works correctly, and intellisense always works as-you-type.
![2018-09-25 21 55 23](https://user-images.githubusercontent.com/3613230/46052895-51bf6200-c10e-11e8-85cb-37860e238100.gif)

Here's a gif with 0.12.7
My cpu is pegged at 100%, intellisense as-you-type doesn't work(have to manually use ctrl+space) _and_ even with that it takes a long time to return results.

![2018-09-25 21 57 12](https://user-images.githubusercontent.com/3613230/46052980-c1355180-c10e-11e8-89cf-faaf128a9bd7.gif)

**Comment by Vorror** [comment]
I know 0.12.1 is pretty old, but it's the only vsix build available. Everything was fine up until today/yesterday, but I can't confirm what build I was before I autoupdated to the latest(0.12.7)

**Comment by Vorror** [comment]
Current veutr settings

```js
  "vetur.validation.template": false,
  "prettier.eslintIntegration": true,
  "vetur.format.defaultFormatter.js": "vscode-typescript",
  "vetur.format.defaultFormatter.html": "js-beautify-html",
  "vetur.format.defaultFormatterOptions": {
    "js-beautify-html": {
      "wrap_line_length": 120,
      "wrap_attributes": "force-aligned"
    }
  }
```

**Comment by octref** [comment]
I made a 0.12.6 package for you. Can you install it and let me know if this causes problem for you? This way I can narrow down the causes.

This is likely caused by upgrading TypeScript to 3.0.3.

https://github.com/vuejs/vetur/releases/tag/v0.12.6

**Comment by Vorror** [comment]
@octref Yep that seems to be the issue

With 0.12.6 everything works correctly
![2018-09-26 01 42 54](https://user-images.githubusercontent.com/3613230/46059570-d2da2180-c12d-11e8-9f8b-bcc292581e44.gif)

I also upgraded again to 0.12.7 afterwards just to make doubly sure I wasn't crazy.

**Comment by aszczepanek** [comment]
This is happend to me too when I opened my Vue project today. Intellisense inside vue files is broken and while 'Loading...' CPU usage goes high.

**Comment by alwonder** [comment]
I found out the problem happens with Typescript 3.0+ in the server. I manually changed TS version in Vue Language Server to 2.9.2 and it started working good.

**Comment by KytoSai** [comment]
I have same isssue :( !!!!

**Comment by geefix** [comment]
Since the other issue (https://github.com/vuejs/vetur/issues/914, seems to me related to this one) is closed i write here. Same problem here, high CPU usage and no autocomplete with 0.12.7.

0.12.6 works fine.

What do you want to know more?

**Comment by dwightjack** [comment]
I can confirm the problem as well: high CPU usage on mac osx, no autocomplete and prettier stopped working (just on .vue files).

Disabled all the extensions but vetur, cleared the cache and reinstalled the extension as well but without any luck.

Rolling back to 0.12.6 fixed the problem.

I'm not an expert about vscode extensions, but if you need any log / detail, feel free to ask for that.

**Comment by octref** [comment]
Sorry for the issue, but I'm having trouble reproducing it. Here are two things that you can do to help me figure out what's going on:

1. Install [this release](https://github.com/vuejs/vetur/releases/tag/0.12.7). It's Vetur 0.12.7 built with TS 2.9.2 (as opposed to the shipped version, TS 3.0.3). See if that makes a difference for you.

2. Use the current Vetur on these projects. Does Vetur become slow? 
  - https://github.com/octref/veturpack
  - https://github.com/chrisvfritz/vue-enterprise-boilerplate
  - https://github.com/vuejs/vue-hackernews-2.0
  - `vue create` with vue-cli. Standard template.

None of the projects above could reproduce the problem for me. If it does for your project, and you cannot share your project with me, can you paste your tsconfig.json or jsconfig.json then? Thanks.

**Comment by alwonder** [comment]
@octref 
As I mentioned above, this is reproduced with TS 3.0+. Your version with TS 2.9.2 works well.

As for projects you listed, they work good on the new version because these projects aren't big enough. I have a working project with ~50k lines, and every typed character costs me 10-15 seconds of waiting. I can't share you my project repo, but I left a VLS log [here](https://github.com/vuejs/vetur/issues/784#issuecomment-424557315). Hope it'll be helpful.

**Comment by KytoSai** [comment]
I tested with 'Vetur 0.12.7 built with TS 2.9.2' . It work fine !!! :) 
Sorry, i'm haven't good english to more support you :( !!

**Comment by dwightjack** [comment]
@octref I can confirm that the build with TS 2.9.2 works for me as well.

With the repositories you provided everything works fine even with the TS3.0+ version. As @alwonder said I guess the issue arises on large projects. To prove that, I removed a bunch of files from the folder (about 400 files mostly unrelated to Vue like static images, JSON files and build scripts)  and the issue disappeared.

Duplicating the same Vue file on a _blank_ project doesn't seem to generate the issue, but I guess there's some sort of cache involved. 
  
Sadly I cannot share the repo I'm working on but here are some more details:

Mac OSX: 10.13.6
Node version in use: 8.11.2
VS Code: 1.27.2 

I have this `jsconfig.json` file in the project root:

```json
{
  "compilerOptions": {
    "baseUrl": "./",
    "paths": {
      "@/*": ["src/*"]
    }
  }
}
```

Let me know if I can provide any other information.

**Comment by geefix** [comment]
For me 0.12.7 /w TS 2.9.2 works slightly better, but still not as good as 0.12.6.

See the difference: 
0.12.6 is faster and in this example _$store_ as a member of _this_ is listed.
![0 12 6](https://user-images.githubusercontent.com/4639153/46337001-bad23880-c62c-11e8-8999-18e3cf476574.gif)

0.12.7 is a bit slower and the list is missing members of _this_
![0 12 7_ts2 9 2](https://user-images.githubusercontent.com/4639153/46337303-814dfd00-c62d-11e8-92c5-8656f6d6b945.gif)


Platform: Win 10 x64
Version: 1.27.2 (user setup)

**Comment by numfin** [comment]
my intelisense is stopping working when i open some large svgs

**Comment by ShaneXie** [comment]
for ppl don't know how to install old version extension on vs code, here is a SO question about it. https://stackoverflow.com/questions/37071388/how-to-install-vscode-extensions-offline

And don't forgot disable _auto updating extensions_

**Comment by octref** [comment]
For those of you who had the problem, can you also try this version which comes with TS 3.1.1?
https://github.com/vuejs/vetur/releases/tag/0.12.7-2

I talked a bit to TS team and there were some perf issues around file watching that is resolved between TS 3.0.3 & 3.1.1.

If this works, I'll update to 3.1.1. If it doesn't, I'll make a new release with TS 2.9.2.

**Comment by ShaneXie** [comment]
@octref Nope, still laggy.

**Comment by octref** [comment]
@ShaneXie Did you also try the TS 2.9.2 version? Was it also buggy so you are using 0.12.6?

**Comment by ShaneXie** [comment]
@octref 0.12.6 works fine for me.

**Comment by lebnor01** [comment]
I've tried the different versions:
1.2.7 that is on marketplace: high cpu usage
1.2.6 relased couple of days ago: no problems
1.2.7 with TS 2.9: seems ok, no high cpu usage
1.2.7 with TS 3.1: high cpu usage

**Comment by octref** [comment]
Thanks again. Based on the feedback, I'll revert TS to 2.9 and make a new release.

**Comment by octref** [comment]
I'm going back to 2.8.4 for now. Before upgrading TS versions again, I'll try to add some logs to TS Language Service so we can figure out what went wrong: https://github.com/vuejs/vetur/issues/922.

**Comment by alwonder** [comment]
@octref Can I ask why not 2.9.2? I just had some troubles with my mixed js-ts project on 2.8.x, and I fixed it installing 2.9.x version on the server, so i'd be happy with 2.9.2.

Anyways, if you leave 2.8.4 eventually and I get some problems, I will create an issue with more details. Thanks a lot for your work!
